Integrations/CofenseTriage/CofenseTriage.py

```diff
@@ -0,0 +1,375 @@
+        headers=HEADERS,
+        params=params,
+        data=body,
+        verify=USE_SSL,
+    )
+    try:
+        if not response.ok:
+            return_error(f'Call to Cofense Triage failed [{response.status_code}] - [{response.text}]')
+
+        elif response.status_code == 206:  # 206 indicates Partial Content, reason will be in the warning header
+            demisto.debug(str(response.headers))
+
+        if raw_response:
+            return response
+        data = response.json() if response.text and response.text != '[]' else {}  # type: Any
+        return data
+
+    except TypeError as ex:
+        demisto.debug(str(ex))
+        return_error(f'Error in API call to Cofense Triage, could not parse result [{response.status_code}]')
+        return {}
+
+
+def test_function() -> None:
+    try:
+        response = requests.get(
+            BASE_URL + '/processed_reports',
+            headers=HEADERS,
+            params="",
+            verify=USE_SSL,
+        )
+        if demisto.getParam("isFetch"):
+            fetch_reports()
+
+        if response.ok:
+            demisto.results('ok')
+
+        else:
+            return_error(
+                f'API call to Cofense Triage failed. Please check Server URL, or authentication related parameters.'
+                f' [{response.status_code}] - {response.reason}')
+
+    except Exception as ex:
+        demisto.debug(str(ex))
+        return_error(f'API call to Cofense Triage failed, please check URL, or authentication related parameters.')
+
+
+def fetch_reports() -> None:
+    # parameters importing
+    start_date, _ = parse_date_range(demisto.getParam('date_range'), date_format=TIME_FORMAT)
+    max_fetch = int(demisto.getParam('max_fetch'))  # type: int
+    params = {
+        'category_id': demisto.getParam('category_id'),
+        'match_priority': demisto.getParam('match_priority'),
+        'tags': demisto.getParam('tags'),
+        'start_date': start_date,
+    }
+
+    # running the API command
+    reports = http_request(url_suffix='/processed_reports', params=params)
+
+    # loading last_run
+    last_run = json.loads(demisto.getLastRun().get('value', '{}'))
+    already_fetched = last_run.get('already_fetched', [])
+
+    # parsing outputs
+    incidents = []
+    for report in reports:
+        if report.get('id') not in already_fetched:
+            category_id, report_id = report.get('category_id'), report['id']
+            report_body = report.pop('report_body')
+            incident = {
+                'name': f"cofense triage report {report_id}: {CATEGORIES.get(category_id, 'Unknown')}",
+                'occurred': report.get('created_at'),
+                'rawJSON': json.dumps(report),
+                'severity': CATEGORIES_SEVERITY.get(category_id, 0)
+            }
+
+            add_attachment(incident, report, report_body, report_id)
+            incidents.append(incident)
+            already_fetched.append(report_id)
```

Comment:
can't we filter just with start date rather than save report ids?

Reply:
start_date is determine by timestamp of the last change made in connection with the reporter (for example, changes to the reporter’s email address, credibility score, or VIP
status or the reporting of an email within the specified date range).
if we will rely on it, we'll get dupes.